### ISSUE TITLE
it is semantically unclear what allow.inf means when we have finite bounds

### ISSUE THREAD

**Issue by berndbischl** [mlr-org/phng#11]
so can the user set lower = 1, upper = 2, allow.inf = TRUE?

**Comment by jakob-r** [comment]
Yes. The user can set this but it will result in values above 2 including Inf to be infeasible.
It is important if `upper = Inf` does not mean that `Inf` is a feasible value.
I think a good documentation solves this issue but if you really want an error message in the given case pls say so.

**Comment by berndbischl** [comment]
> The user can set this but it will result in values above 2 including Inf to be infeasible.

huh? no.

**Comment by berndbischl** [comment]
1) first of all, your last commit broke (reasonable) unit tests, where i purposefully excluded this issue here

2) i know what allow.inf is intended to do, basically [0, inf) vs [0, inf]

3) for this case one has 2 options:
> so can the user set lower = 1, upper = 2, allow.inf = TRUE?
a) immediatly throw an arg-check-error
b) allow [1,2] union inf

i thought b) would be ok, because we also have special.vals, but a) it better.
but needs to be documented, and unit tested
